src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -161,6 +161,22 @@ public virtual IEnumerable<PackageResult> List(ChocolateyConfiguration config)
                 }
             }
 
+            if ((config.ListCommand.ApprovedOnly || config.ListCommand.DownloadCacheAvailable || config.ListCommand.NotBroken) && config.RegularOutput)
+            {
+                // This warning has been added here, to provide context to the user, and a follow up issue
+
+                // has been added here: https://github.com/chocolatey/choco/issues/3139 to address the actual
+                // issue that causes this warning to be required.
+                this.Log().Warn(@"
+Starting vith Chocolatey CLI v2.0.0, changes have been made to the
+`choco search` command which means that filtering of packages using the
+--approved-only, --download-cache, and --not-broken options are not
+performed within Chocolatey CLI. These are instead performed by the
+Chocolatey Community Repository. As a result, it is possible that
+incomplete package lists are returned from a command that uses these
+options.");

```

Comment:
Isn't this the opposite?

It is now performed by the Chocolatey CLI, not by Chocolatey Community Repository?

Reply:
Yes, I believe you are right.  I have fixed this up, can you take another look?